fmralign/template_alignment.py

```diff
@@ -0,0 +1,349 @@
+from sklearn.base import BaseEstimator, TransformerMixin
+import numpy as np
+from joblib import Parallel, delayed
+from sklearn.externals.joblib import Memory
+from nilearn.image import index_img
+from nilearn.input_data.masker_validation import check_embedded_nifti_masker
+from fmralign.pairwise_alignment import PairwiseAlignment
+
+
+def euclidean_mean(imgs, masker, scale_template=False):
```

Comment:
if scale_template is False, then this is nothing else than Nielarn's `mean_img` I am a bit afraid of using `scale_template=True`, since the behavior becomes really different from 'euclidean mean'. 
At least , I'd rather call the function rescaled_mean to make its behavior clearer to users.

Reply:
Also, this should be a private function (_rescaled_mean)

Reply:
Your first comment is inexact, I recoded the function because I needed it :
mean_img docstring reads "Compute the mean of the images (in the time dimension of 4th dimension)" whereas euclidean_mean can handle lists of 4d images and compute their mean in the subject dimension which is the behavior I use by default.

Ok to rename it rescaled_mean, even though this name reflects a behavior that is not used most of the time.

Not sure it should be private. In every example / experiment with template, this will make the baseline mean of images across subjects

Reply:
But why not call `mean_img` in those cases ?

Reply:
... see my previous comment, they don't have the same behavior on lists of 4D Niimgs.

`mean_img` will return a 3D Niimg as nilearn docs state :
> Note that if list of 4D images are given, the mean of each 4D image is computed separately, and the resulting mean is computed after.

Unlike that, `euclidian_mean` will return a 4D Niimgs (just averaging across subject **i.e. the axis 0** from the list)  : 
```
def euclidean_mean(imgs, masker, scale_template=False):
    masked_imgs = [masker.transform(img) for img in imgs]
    average_img = np.mean(masked_imgs, axis=0)
```

Reply:
anyway, I took your advices